src/fu-tool.c

```diff
@@ -1317,10 +1317,21 @@ fu_util_update_all(FuUtilPrivate *priv, GError **error)
 	for (guint i = 0; i < devices->len; i++) {
 		FwupdDevice *dev = g_ptr_array_index(devices, i);
 		FwupdRelease *rel;
-		const gchar *device_id;
+		const gchar *device_id = fu_device_get_id(dev);
 		g_autoptr(GPtrArray) rels = NULL;
 		g_autoptr(GError) error_local = NULL;
-
+		gboolean dev_skip_byid = TRUE;
+
+		/* only process particular DEVICE-ID or GUID if specified */
+		for (guint idx = 0; idx < g_strv_length(values); idx++) {
+			const gchar *guid = values[idx];
```

Comment:
this variable shouldn't be called guid since it can be used for either device ID or guid.  Maybe tmp instead?